### PR TITLE
Share geolocation demo

### PR DESCRIPTION
**Pull request by markus-kohler**
### Features
- Listen to "DM*" channel for any direct messages sent across the chat. If a message is for the login user, display the message on the map in a little pop up
- Publish current location message across PubNub into "Public.location-chat"
- Documentation for Geolocation.js

### Notes
- Moved Map functionality into map.js

### CODE REVIEW

**Comment by darryncampbell-pubnub** [comment]
Looking good Markus, I have a few comments:
* I'm getting an error whenever I enter a new location (in the console window)
* It feels like I should be able to click on the previous locations in the travel history (they have an animation when I click them) but nothing happens after I click them.  Is it possible to remove the animation?
* When I hit 'share' it should take me to the correct channel in chat (see my other review comment in the code related to this)
* I'm unable to get the popup to appear when I send a message.
I took a screencapture of all the above, which I'll share on Slack with you
* Is it possible to show an error if I select 'use current location', then do not allow the browser to use my location

File: web/geolocation/message.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ */
+
+//  Handler for the PubNub message event
+async function messageReceived(payload){
+	// If the channel indicates it is a direct message
+	if(payload.channel.includes(pubnub.getUUID())){
+		if(mark.hasOwnProperty(payload.publisher)){
+			// Show Message in a popup bubble above users marker
+			try{
+				var infoMessage = "<b>Message delivered via PubNub</b> <br></br>";
```
[the remark this answers is not in the document]
Comment (markus-kohler) [reply web/geolocation/message.js:14]: Yes I did, I will update that.

File: web/geolocation/geolocation.html [hunk 2]
```diff
@@ -40,18 +38,12 @@ <H3 id='header-subheading' class="text-center"> </H3>
...
-                    </div>
-                    <div class="col-auto">
-                        <span id="geo-channel-id">GEO-Channel ID</span><span id="channel-label" class="custom-bubble">XXXXX</span>
                     </div>
                 </div>
                 <br>
                 <div class="row">
                     <div id="map-canvas-div">
-                        <div id="map-canvas" style="width:100%;height:100%"></div>
+                        <div id="map-canvas" style="width:100%; height:100%"></div>
```
[the remark this answers is not in the document]
Comment (darryncampbell-pubnub) [reply web/geolocation/geolocation.html]: Styles have been added

**Review by darryncampbell-pubnub — APPROVED** [review]
All looks good

**Pull request merged**